WooCommerce/Classes/ViewRelated/Blaze/BlazeCampaignList/BlazeCampaignListView.swift

```diff
@@ -63,6 +57,81 @@ final class BlazeCampaignListHostingController: UIHostingController<BlazeCampaig
     }
 }
 
+// MARK: - Navigation related
+private extension BlazeCampaignListHostingController {
+    /// Handles navigation to the different campaign creation views.
+    func navigateToCampaignCreation() {
+        switch viewModel.createCampaignDestination {
+        case .productSelector:
+            navigateToBlazeProductSelector(source: .campaignList)
+        case .campaignForm(let productID):
+            navigateToNativeCampaignCreation(source: .campaignList, productID: productID)
+        case .webviewForm(let productID):
+            navigateToWebCampaignCreation(source: .campaignList, productID: productID)
+        case .noProductAvailable:
+            break // TODO 11685: add error alert.
+        }
+    }
+
+    /// Handles navigation to the webview Blaze creation
+    func navigateToWebCampaignCreation(source: BlazeSource, productID: Int64? = nil) {
+        let webViewModel = BlazeWebViewModel(siteID: viewModel.siteID,
+                source: source,
+                siteURL: viewModel.siteURL,
+                productID: productID) { [weak self] in
+            self?.handlePostCreation()
+        }
+        let webViewController = AuthenticatedWebViewController(viewModel: webViewModel)
+        self.navigationController?.show(webViewController, sender: self)
+        viewModel.didSelectCreateCampaign(source: source)
+    }
+
+    /// Handles navigation to the native Blaze creation
+    func navigateToNativeCampaignCreation(source: BlazeSource, productID: Int64) {
+        let campaignCreationFormViewModel = BlazeCampaignCreationFormViewModel(siteID: viewModel.siteID,
+                                                                               productID: productID,
+                                                                               onCompletion: { })
+        let controller = BlazeCampaignCreationFormHostingController(viewModel: campaignCreationFormViewModel)
+
+        // This function can be called from product selector, which is handled by BlazeNavigationController.
+        // In this case, we need to show the controller from BlazeNavigationController.
+        // Otherwise, we show it from the current navigation controller.
+        if blazeNavigationController.presentingViewController != nil {
+            blazeNavigationController.show(controller, sender: self)
+        } else {
+            self.navigationController?.show(controller, sender: self)
```

Comment:
`self` is not needed.

Reply:
This is also removed completely as part of refactor.